activemodel/lib/active_model/validations/length.rb

```diff
@@ -15,7 +15,8 @@ class LengthValidator < EachValidator # :nodoc:
       def initialize(options)
         if range = (options.delete(:in) || options.delete(:within))
           raise ArgumentError, ":in and :within must be a Range" unless range.is_a?(Range)
-          options[:minimum], options[:maximum] = range.min, range.max
+          options[:minimum] = range.min if range.begin
+          options[:maximum] = (range.exclude_end? ? range.end - 1 : range.max) if range.end
```

Comment:
It looks like Ruby 3.0 can handle `(..10).max`, but Ruby 2.7 can't: https://buildkite.com/rails/rails/builds/86647#c7276773-3079-4698-976f-ae955eed802c/1165-1176.  (Glad we added that test case! :sweat_smile:)

I suppose we can just use `range.end` for both branches.